node/caching-compiler.js

```diff
@@ -17,27 +17,32 @@ const reifyVersion = require("../package.json")
 
 const DEFAULT_CACHE_DIR = ".reify-cache";
 
-exports.compile = (content, filename) => {
-  if (filename === "repl") {
+exports.compile = (content, options) => {
+  options = options || Object.create(null);
+
+  if (options.filename === "repl") {
     // Treat the REPL as if there was no filename.
-    filename = null;
+    options.filename = null;
```

Comment:
We shouldn't mutate the `options` object. It can be avoided by using `options = Object.assign(Object.create(null), options)` or by referencing the options as one off vars.